src/layout.cpp

```diff
@@ -77,4 +83,19 @@ Status Layout::validate() const {
     return StatusCode::OK;
 }
 
+std::optional<Layout> Layout::createIntersection(const Layout& other) const {
+    Layout layout;
+    if (*this != other) {
```

Comment:
To make the nesting less annoying (less nested) you can first check if `if(*this == other)` and return immediately if so. Then the next lines can be reduced with 1 level of nesting.